src/xrCore/XML/XMLDocument.cpp

```diff
@@ -30,17 +86,18 @@ void ParseFile(pcstr path, CMemoryWriter& W, IReader* F, XMLDocument* xml)
 
     while (!F->eof())
     {
-        F->r_string(str, sizeof str);
-
-        // Skip any spaces or tabs
-        pcstr begin_of_include = str;
-        while (*begin_of_include != '\0' && std::isblank(*begin_of_include))
-            ++begin_of_include;
+        string4096 str;
+        if (!F->try_r_string(str, sizeof(str)))
+        {
+            R_ASSERT3(!fatal, "XML file[%s] parsing failed. Line is too long (>= 4096)", path);
+            Msg("! XML file[%s] parsing failed. Line is too long (>= 4096)", path);
```

Comment:
This is not needed, because R_ASSERT message is already being included in the log
```suggestion
```